src/d3d12/d3d12_device.cpp

```diff
@@ -125,6 +125,59 @@ namespace wr::d3d12
 			device->m_adapter = (IDXGIAdapter4*)adapter;
 		}
 
+		void QueryForOptionalFormats(Device* device)
+		{
+			D3D12_FEATURE_DATA_D3D12_OPTIONS feature_data;
+			ZeroMemory(&feature_data, sizeof(feature_data));
+			HRESULT hr = device->m_native->CheckFeatureSupport(D3D12_FEATURE_D3D12_OPTIONS, &feature_data, sizeof(feature_data));
+
+			if (SUCCEEDED(hr))
+			{
+				// TypedUAVLoadAdditionalFormats contains a Boolean that tells you whether the feature is supported or not
+				if (feature_data.TypedUAVLoadAdditionalFormats)
+				{
+					// Can assume �all-or-nothing� subset is supported (e.g. R32G32B32A32_FLOAT)
+					// Cannot assume other formats are supported, so we check:
+
+					auto CheckAndSetFormat = [](Device* device, DXGI_FORMAT format)
```

Comment:
function would be preferable here. No real benefit of using a lambda in this case.